api/src/wfl/module/covid.clj

```diff
@@ -198,24 +173,22 @@
    :table   :dataset_table
    :column  :table_column_name})
 
-(defn ^:private peek-tdr-source-queue [{:keys [queue] :as _source}]
-  (let [query "SELECT * FROM %s ORDER BY id ASC LIMIT 1"]
-    (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-      (first (jdbc/query tx (format query queue))))))
-
-(defn ^:private pop-tdr-source-queue [{:keys [queue] :as source}]
-  (if-let [{:keys [id] :as _snapshot} (peek-queue! source)]
-    (let [query "DELETE FROM %s WHERE id = ?"]
-      (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-        (jdbc/execute! tx [(format query queue) id])))
-    (throw (ex-info "No snapshots in queue" {:source source}))))
+(defn ^:private create-tdr-source [tx id request]
+  (let [create  "CREATE TABLE %s OF TerraDataRepoSourceDetails (PRIMARY KEY (id))"
```

Comment:
For my own understanding, I'm not sure if this will create the `id` field with auto increment feature (e.g. `SERIAL` or `GENERATED AS IDENTITY`), since it's not defined in our `CREATE TYPE TerraDataRepoSourceDetails` composition type?

Reply:
You could try an `"ALTER TABLE %s ALTER COLUMN id ADD GENERATED ALWAYS AS IDENTITY"`
after the `"CREATE TABLE OF %s"`.  That might do the trick.
See https://www.postgresql.org/docs/11/sql-altertable.html

Reply:
I'll throw out there that I moved this code block but I didn't write it initially, tagging in @ehigham .

(Lesson learned: touch not what you don't understand / aren't willing to fix :) )